### PR TITLE
Create storage abstraction

### PR DESCRIPTION
**Pull request by KevinDyer**
Went on a crusade to move all dbQuery functions in a their own directory. The purpose is to create a storage layer abstraction so a new storage layer could be used at some point (also cleaner code). I'm not the happiest at myself, this PR could use a lot of tests...

### CODE REVIEW

**Comment by KevinDyer** [comment]
This builds on the [fix PR](https://github.com/graffiticode/graffiticode/pull/11)

**Comment by jeffdyer** [comment]
My tweaks causes merge conflicts. I'll do a manual merge later today.

**Comment by KevinDyer** [comment]
I was able to get it rebased quickly I think it is good to go

**Comment by jeffdyer** [comment]
Nice refactor. I'll merge into master and push to staging: gc.s.acx.ac (staging.graffiticode.com) for sanity testing before pushing to prod.

File: app.js [hunk 1]
```diff
@@ -1123,66 +1124,102 @@ app.put('/code', (req, res) => {
...
+  }
+  limit = Number.parseInt(limit);
+  if (!Number.isInteger(limit)) {
+    return res.status(400).json({
+      success: false,
+      errors: ['limit must be an integer'],
+      data: null,
+    });
+  }
+  if(limit < 1 || limit > 100) {
```
Comment (jeffdyer) [inline app.js]: We are often looking for many more than 100 items. What's the rationale for this restriction?

File: app.js [hunk 2]
```diff
@@ -1123,66 +1124,102 @@ app.put('/code', (req, res) => {
...
+      });
     } else {
-      console.log("GET /items " + rows.length + " found, " + (new Date - t0) + "ms");
-      res.send(rows);
+      console.log(`GET /items ${result.rows.length} found, ${(new Date - t0)}ms`);
+      res.status(200).json({
+        success: true,
+        errors: [],
+        data: result.rows,
+      });
```
Comment (jeffdyer) [inline app.js]: I like this protocol. Previously we were just returning `result.rows`, so clients have to be updated to handle the new response format. Once case is in `./src/archive-view.js`. Various compilers also hit this endpoint. I'll update known callers.

**Pull request merged**